mscore/importmxmlpass2.cpp

```diff
@@ -4449,16 +4247,41 @@ Note* MusicXMLParserPass2::note(const QString& partId,
             return 0;
             }
 
-      TDuration duration = determineDuration(rest, type, mnd.dots(), dura, measure->ticks());
+      // start time for note:
+      // - sTime for non-chord / first chord note
+      // - prevTime for others
+      auto noteStartTime = chord ? prevSTime : sTime;
+      const auto timeMod = mnd.timeMod();
+
+      // determine tuplet state, used twice (before and after note allocation)
+      MxmlTupletFlags tupletAction;
+
+      // handle tuplet state for the previous chord or rest
+      if (!chord && !grace) {
+            auto& tuplet = tuplets[voice];
+            auto& tupletState = tupletStates[voice];
+            tupletAction = tupletState.determineTupletAction(mnd.dura(), timeMod, notations.tupletDesc().type, mnd.normalType(), missingPrev, missingCurr);
+            if (tupletAction & MxmlTupletFlag::STOP_PREVIOUS) {
+                  // tuplet start while already in tuplet
```

Comment:
Couldn't there be nested tuplets? MuseScore allows that (Silbelius [seems to support them](https://www.scoringnotes.com/tips/fun-with-nested-tuplets/) too) that so musicxml file can theoretically contain nested tuplets rather than incompletely written tuplets. Or are they handled elsewhere with this patch, or not allowed by musicxml perhaps?

Reply:
Ah, sorry, I see them marked unsupported, then my question is not relevant
https://github.com/musescore/MuseScore/blob/dbd64797ce6b30c29face555d2dd841e0c5a89e6/mscore/importmxmlpass1.cpp#L2777